docs/technical_tutorial/apollo_vehicle_adaption_tutorial_cn.md

```diff
@@ -0,0 +1,935 @@
+ 乘用车标准：http://apollo-homepage.bj.bcebos.com/Apollo_by_wire_requirement.xlsx ；
+微型车标准：https://apollohomepage.bj.bcebos.com/Apollo_by_wire_requirement_micro_car.xlsx
+车辆线控标准规定了车辆底盘信号的需求，也规定了每一帧控制信号在底盘的响应时间要求。车辆底盘信号不仅需要在功能上符合要求，也需要在性能上符合控制要求，这样才能满足符合Apollo线控标准的车辆控制底盘。
+ - 开发者在改造车辆底盘为线控底盘时，要参考车辆线控标准满足控制的性能要求，如转向信号要求响应时间在100ms内，最大超调量满足满足具体要求等。请开发者仔细阅读线控列表内的性能要求，在Apollo车辆认证时，也会对底盘的性能进行相应测试。
+
+在实际开发DBC的过程中，每个信号定义的细节也有具体的要求，这样做为了节省开发者时间，同时提高车辆认证的效率，需要开发者对每个信号要重点理解和注意。下面是Apollo对车辆底盘信号的需求如下表所示：
+![apollo_by_wire_requirement1](images/vehicle_adaption_tutorial/apollo_by_wire_requirement1.png)
+![apollo_by_wire_requirement2](images/vehicle_adaption_tutorial/apollo_by_wire_requirement2.png)
+![apollo_by_wire_requirement3](images/vehicle_adaption_tutorial/apollo_by_wire_requirement3.png)
+
+车辆的底盘控制是Apollo与底盘通信的基础，每一个控制信号都需要可靠和准确。人工接管和越界处理是保证在测试和使用时随时接管，确保突发情况发生时的行车安全。除了线控列表内对信号的详细要求外，**开发者还需要格外注意横向和纵向使能信号的独立**，**底盘接管逻辑等特殊要求**，下面对信号的功能特殊要求做详细说明。
+
+#### 线控转向
+1. 转向使能信号
+
+（1）转向使能信号是控制车辆进入或退出转向自动驾驶模式，该使能信号必须是通过上升沿触发方式的信号，使能信号原则上是1个信号；有些开发者若有不同信号同时控制，才能使能转向进入自动驾驶状态，需要开发者在底层自己集成相关使能信号的逻辑控制，给Apollo使能信号原则上必须只有1个。
+
+（2）转向自动驾驶模式必须独立，使能信号必须独立，不与纵向的其他模块（如驱动、制动、档位、驻车等）使能信号相关联，需要满足：单独使能进入转向自动驾驶模式，纵向所处驾驶模式不发生变化，下发方向盘转角或转速控制信号，车辆能够响应相应的转角和转速；单独使能退出转向自动驾驶模式，纵向所处驾驶模式不发生变化，下发方向盘转角或转速控制信号，车辆不响应；
+
+（3）如果横向被接管，再次进入自动驾驶模式时，需要通过上升沿触发信号进入自动驾驶。
+
+2. 转向控制信号
+
+目标方向盘转角：控制方向盘转角的信号，单位是度（deg），控制方向盘（逆时针）左转为正值，控制方向盘（顺时针）右转为负值。
+
+目标方向盘转速：多数车辆无法开放次控制信号接口，建议将方向盘转速设置为最大值，提高系统响应时间。
+
+3. 转向反馈信号
+
+方向盘实际转角反馈：反馈方向盘实际转角，要求反馈准确，左转反馈为正值，右转反馈为负值。
+
+转向驾驶模式反馈：反馈车辆当前所处的转向驾驶模式，至少需要3种模式：自动驾驶模式，手动驾驶模式，被人工接管模式。
+ 
+故障信息反馈：反馈转向控制系统是否存在故障。
+
+4. 人工接管处理
+
+（1）对于乘用车/商用车，当人为施加的方向盘扭矩大于其接管门限时，车辆各个模块均退出自动驾驶，此时转向驾驶模式反馈被接管；对于底盘小车等，当小车底盘接收到遥控器控制车辆转向的指令时，车辆各个模块均退出自动驾驶；转向接管需要独立，此时转向驾驶模式反馈被人工接管模式，驱动驾驶模式、制动驾驶模式此时应反馈手动驾驶模式；
+
+（2）当转向自动驾驶被人工接管时，所有模块都退出自动驾驶模式。
+
+5. 越界处理
+
+越界处理的原则是越界拒绝执行，并退出自动驾驶模式。
+当Apollo下发控制指令超过原控制指令的定义范围时，此时称为越界。发生越界时，转向需要退出自动驾驶模式。
+如定义目标方向盘转角最大范围为-500deg~500deg，若Apollo下发的转角指令不在此范围时，底盘需要对此越界拒绝执行，并退出自动驾驶模式。
+
+#### 线控驱动
+
+1. 驱动使能：
+
+（1）驱动使能信号是控制车辆进入或退出驱动自动驾驶模式，该使能信号必须是通过上升沿触发方式的信号，**使能信号原则上是1个信号**；有些开发者若有不同信号同时控制，才能使能转向进入自动驾驶状态，需要开发者在底层自己集成相关使能信号的逻辑控制，给Apollo使能信号原则上必须只有1个。
+
+（2）驱动使能信号必须独立，不与转向模块使能信号相关联，需要满足：单独使能进入纵向自动驾驶模式，转向所处驾驶模式不发生变化，下发驱动控制信号，车辆能够响应相应的控制命令；单独使能退出驱动自动驾驶模式，转向所处驾驶模式不发生变化，下发驱动控制信号，车辆不响应；
+
+（3）**原则上驱动使能信号与制动、档位、驻车使能相互独立。**根据以往的实际经验，有些车辆在纵向驱动和制动使能信号归为一个信号，这样的特殊情况需要开发者额外注意，确保纵向的使能一定要与横向使能分开并独立，纵向各模块如果相互关联，则使能需要驱动、制动、档位、驻车一同进去自动驾驶模式，不使能一同退出自动驾驶模式。
+
+（4）如果横向被接管，再次进入自动驾驶模式时，需要通过上升沿触发信号进入自动驾驶。
+
+2. 驱动控制信号：
+
+目标加速踏板位置：控制量是加速踏板的位置百分比，范围是0~100%，100%代表加速踏板踩到最大位置。
+
+车辆目标纵向加速度和车辆目标驱动扭矩是车辆控制信号可选项。
+
+3. 驱动反馈信号：
+
+驾驶模式：反馈车辆当前所处的驱动驾驶模式，至少需要3种模式：自动驾驶模式，手动驾驶模式，被人工接管模式。
+
+加速踏板位置：需准确反馈当前车辆实际踏板位置。
+
+纵向加速度：需准确反馈当前车辆的纵向加速度。
+
+车速：需准确反馈当前车辆车速，Apollo程序内需求的车速是m/s，**一般车辆反馈的车速单位是km/h，此时需要在DBC转换为适配代码后单独进行单位转化**。
+
+轮速：对于乘用车，需提供相关轮速信号，轮速大小，单位km/h。如果有条件，可提供车轮运动方向反馈，Value Description按照以下顺序0x00，FORWARD；0x01，BACKWARD；0x02，STANDSTILL；0x03，INVALID。
```

Comment:
for the wheel_encoder, could you double check if we assume the unit `km/h`? I thought we enforce `SI` unit.

Reply:
Checking the apollo by wire requirement, we assume the unit kph(km/h). For some OEMs, they want to output the wheel speed with the km/h unit because they are using the kph in their vehicles. Therefore, we need to transform the unit for SI unit in protocol.